src/components/TodoInfo/TodoInfo.tsx

```diff
@@ -1 +1,51 @@
-export const TodoInfo = () => {};
+import cn from 'classnames';
+import users from '../../api/users';
+import { UserInfo } from '../UserInfo';
+
+interface Props {
+  title: string,
+  userId: number,
+  id: number,
+  completed: boolean,
+}
+
+interface User {
+  id: number,
+  name: string,
+  username: string,
+  email: string,
+}
+
+function getUserById(arr: User[], id: number) {
+  return arr.find(item => item.id === id);
+}
+
+export const TodoInfo: React.FC<Props> = ({
+  id,
+  userId,
+  title,
+  completed,
+}) => {
+  const user = getUserById(users, userId);
+  const email = user?.email;
+  const name = user?.name;
+
+  return (
+    <section
+      className="TodoList"
+    >
+      <article
+        data-id={id}
+        className={cn('TodoInfo', {
+          'TodoInfo--completed': completed,
+        })}
+      >
+        <h2 className="TodoInfo__title">
+          {title}
+        </h2>
+
+        <UserInfo email={email} name={name} />
```

Comment:
user can be undefined, so let's handle this case here

because UserInfo doesn't need to know if the user exists or not, it just needs to output it

Reply:
![image](https://github.com/mate-academy/react_add-todo-form/assets/55200530/9f43881f-e8b5-414d-8307-990d2cd25fa0)
still not fixed